io/pci/dlpar.py

```diff
@@ -0,0 +1,257 @@
+        self.login(self.hmc_ip, self.hmc_username, self.hmc_pwd)
+        self.run_command("uname -a")
+        cmd = 'lshwres -r io -m ' + self.server + \
+              ' --rsubtype slot --filter lpar_names=' + self.lpar_1 + \
+              ' -F drc_index,lpar_id,drc_name | grep -i %s ' % self.loc_code
+
+        output = self.run_command(cmd)
+        self.drc_index = output[-1].split(',')[0]
+        self.lpar_id = output[-1].split(',')[1]
+        self.log.info("lpar_id : %s, loc_code: %s, drc_index: %s",
+                      self.lpar_id, self.loc_code, self.drc_index)
+
+    def login(self, ip, username, password):
+        '''
+        SSH Login method for remote server
+        '''
+        p = pxssh.pxssh()
+        # Work-around for old pxssh not having options= parameter
+        p.SSH_OPTS = p.SSH_OPTS + " -o 'StrictHostKeyChecking=no'"
+        p.SSH_OPTS = p.SSH_OPTS + " -o 'UserKnownHostsFile /dev/null' "
+        p.force_password = True
+
+        p.login(ip, username, password)
+        p.sendline()
+        p.prompt(timeout=60)
+        # Ubuntu likes to be "helpful" and alias grep to
+        # include color, which isn't helpful at all. So let's
+        # go back to absolutely no messing around with the shell
+        p.set_unique_prompt()
+        p.prompt(timeout=60)
+        self.pxssh = p
+
+    def run_command(self, command, timeout=300):
+        '''
+        SSH Run command method for running commands on remote server
+        '''
+        self.log.info("Running the command on hmc %s", command)
+        c = self.pxssh
+        c.sendline(command)
+        c.expect("\n")  # from us
+        c.expect(c.PROMPT, timeout=timeout)
+        output = c.before.splitlines()
+        c.sendline("echo $?")
+        c.prompt(timeout)
+        return output
+
+    def test(self):
+        '''
+        Test all services and start the DLPAR of drc device
+        DLPAR remove, add and move operations from lpar_1 to lpar_2
+        '''
+        try:
+            process.run("startsrc -g rsct", shell=True, sudo=True)
+        except CmdError as details:
+            self.log.debug(str(details))
+            self.fail("Command startsrc -g rsct failed")
+
+        try:
+            process.run("startsrc -g rsct_rm", shell=True, sudo=True)
+        except CmdError as details:
+            self.log.debug(str(details))
+            self.fail("Command startsrc -g rsct_rm failed")
+
+        output = process.system_output("lssrc -a", ignore_status=True,
+                                       shell=True, sudo=True)
+        if "inoperative" in output:
+            self.fail("Failed to start the rsct and rsct_rm services")
+
+        self.dlpar_remove()
+        self.dlpar_add()
+        self.dlpar_move()
+
+    def dlpar_remove(self):
+        '''
+        dlpar remove operation
+        '''
+        output = self.changehwres(self.server, 'r', self.lpar_id, self.lpar_1,
```

Comment:
since `changehwres()` does not need to return, this variable `output` is not needed here and other places.

Reply:
correct